src/internal/m365/onedrive/permission.go

```diff
@@ -344,13 +417,41 @@ func RestorePermissions(
 
 	ctx = clues.Add(ctx, "permission_item_id", itemID)
 
+	previousLinkShares, err := computePreviousLinkShares(ctx, itemPath, caches.ParentDirToMeta)
+	if err != nil {
+		return clues.Wrap(err, "previous link shares")
+	}
+
+	lsAdded, lsRemoved := metadata.DiffLinkShares(previousLinkShares, current.LinkShares)
+
+	didReset, err := UpdateLinkShares(
```

Comment:
it looks like restoring permissions is sensitive to whether we restore link shares first or the old style permissions (based on the comment in the `if didReset` block). Can you leave a comment in the code stating that please?